pubsub/google/cloud/pubsub_v1/subscriber/futures.py

```diff
@@ -29,10 +29,18 @@ class Future(futures.Future):
     Args:
         policy (~.pubsub_v1.subscriber.policy.base.BasePolicy): The policy
             that creates this Future.
+        event_factory (Optional[Callable[[], Any]]): An event factory, expected
+            to take no arguments and return an event with the same interface as
+            :class:`threading.Event`. This is provided so that callers
+            with different concurrency models (e.g. ``threading`` or
+            ``multiprocessing``) can wait on an event that is compatible
+            with that model. The ``wait()`` and ``set()`` methods will be
+            used on the returned event. The default is
+            :class:`threading.Event`.
     """
-    def __init__(self, policy):
+    def __init__(self, policy, *args, **kwargs):
```

Comment:
@lukesneeringer Would you rather I use `*a, **k` here or just be explicit with the `event_factory` argument? It may make more sense to make `event_factory=None` the default in the latter case.

Reply:
Will every subclass need `event_factory`?

Reply:
Well the subclass needs to be able to pass it through or it's not a useful argument. The point is that one caller could create a `Future(threading.Event)` and another could pass a `Future(multiprocessing.Event)` (or likely `manager.Event`).

(In other words, this isn't publisher or subscriber specific, it's concurrency model specific.)

Reply:
FWIW I struggled with three choices for that input:

- `completed` (`None` or an already instantiated `Event`)
- `event_class` (a class with a constructor that takes no arguments)
- `event_factory` (alternate name: `make_event`)